File: backend/server/model/address.go
package model

type Province struct {
	ID   string `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	Name string `json:"name"`

	Districts []District `json:"districts,omitempty"`
}

type District struct {
	ID         string  `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	Name       string  `json:"name"`
	ProvinceID *string `gorm:"type:uuid" json:"province_id"`

	Province *Province `json:"province,omitempty"`
	Wards    []Ward    `json:"wards,omitempty"`
}

type Ward struct {
	ID         string  `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	Name       string  `json:"name"`
	DistrictID *string `gorm:"type:uuid" json:"district_id"`

	District *District `json:"district,omitempty"`
}

type Address struct {
	ID        string  `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	UserID    *string `gorm:"type:uuid" json:"user_id"`
	Fullname  string  `gorm:"type:varchar(255)" json:"fullname"`
	Phone     string  `gorm:"type:varchar(10)" json:"phone"`
	WardID    *string `gorm:"type:uuid" json:"ward_id"`
	Detail    string  `json:"detail"`
	IsDefault bool    `json:"is_default"`

	User *User `json:"user,omitempty"`
	Ward *Ward `json:"ward,omitempty"`
}

File: frontend/ReSell/app/src/main/java/com/example/resell/model/Address.kt
package com.example.resell.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Address(
    @Json(name = "id")
    val id: String,

    @Json(name = "user_id")
    val userId: String,

    @Json(name = "ward_id")
    var wardId: String,

    @Json(name = "detail")
    var detail: String,

    @Json(name = "is_default")
    var isDefault: Boolean,

    @Json(name = "ward")
    val ward: Ward? = null,
    @Json(name = "fullname")
    val fullname: String? = null,
    @Json(name = "phone")
    val phone: String? = null,

)
